bwt/bwt.go

```diff
@@ -0,0 +1,227 @@
+package bwt
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+const nullChar = "0"
+
+// BWT Burrow Wheeler Transform
```

Comment:
I love this opening explanation. Makes it very clear why this is necessary.

Reply:
You've probably put this somewhere but I think the naive implementation as pseudocode may be good to explain the very simple core of the algo and help demystify it.

The stuff from Langmead should also be cited somewhere in the code itself.

Reply:
For sure! I'll do something similar with what I did with the wavelet tree to illustrate things like:
* the BWT itself
* how the first and last column have characters of the same rank in the same order that enables all the magic
* LF mapping
* Compressing "runny" sequences like the first column.

I'd be happy to cite him. Are there any examples of citation that exist in Poly? If not, is there a preferred way to do that?

Reply:
I Cited him at the end of the BWT's top comment.